pkg/backup/backupsink/sink_utils.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package backupsink
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/builtins"
+	"github.com/cockroachdb/cockroach/pkg/storage"
+	"github.com/cockroachdb/errors"
+)
+
+// ElidedPrefix returns the prefix of the key that is elided by the given mode.
+func ElidedPrefix(key roachpb.Key, mode execinfrapb.ElidePrefix) ([]byte, error) {
```

Comment:
ElidePrefix doesn't really belong in file sink as we use this func during restore, where no file sink is involved. If the key rewriter was in its own package, which it really should be (we use it in pcr and ldr), this ElidedPrefix util would live there. Fine to keep this as is, but i'm making a mental note for myself to move the key rewriter into its own package once this merges.